### PR TITLE
Unit tests for interior vehicle data management

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Provide Unit tests

### Summary
Have added some new test cases for interior vehicle data management and some fixes

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc [hunk 1]
```diff
@@ -180,37 +178,52 @@ TEST_F(GetInteriorVehicleDataRequestTest,
...
       .WillOnce(Return(true));
+  // Act
+  command->Run();
+}
 
+TEST_F(GetInteriorVehicleDataRequestTest,
+       Execute_ExpectCorrectMessageSentToHMI_SupscribeAppNoCache) {
+  // Arrange
+  InitialExpectations();
+  auto module_type = mobile_apis::ModuleType::RADIO;
```
Comment (ZhdanovP) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc]: @LitvinenkoIra It's seems better to move `module_type` in GetInteriorVehicleDataRequestTest and make const. I see that we are using `mobile_apis::ModuleType::RADIO` in all tests. We can set this value once in class.

Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc]: @ZhdanovP fixed
fe38fb0

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc [hunk 2]
```diff
@@ -95,41 +96,43 @@ class GetInteriorVehicleDataRequestTest
         .WillByDefault(Return(rc_app_extention_));
     ON_CALL(*mock_app2_, QueryInterface(_))
         .WillByDefault(Return(rc_app_extention2_));
+  }
+
+  MessageSharedPtr CreateBasicMessage() {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc]: @LitvinenkoIra add comment

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc [hunk 3]
```diff
@@ -95,41 +96,43 @@ class GetInteriorVehicleDataRequestTest
...
+    MessageSharedPtr message = CreateMessage();
+    (*message)[application_manager::strings::params]
+              [application_manager::strings::function_id] =
+                  mobile_apis::FunctionID::GetInteriorVehicleDataID;
+    (*message)[application_manager::strings::params]
+              [application_manager::strings::connection_key] = kConnectionKey;
+    (*message)[application_manager::strings::params]
+              [application_manager::strings::connection_key] = kAppId;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc]: @LitvinenkoIra specify in doxygen that this message is for app1

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc [hunk 4]
```diff
@@ -95,41 +96,43 @@ class GetInteriorVehicleDataRequestTest
...
+  void InitialExpectations() {
     ON_CALL(app_mngr_, hmi_interfaces())
         .WillByDefault(ReturnRef(mock_hmi_interfaces_));
-
     ON_CALL(
         mock_hmi_interfaces_,
         GetInterfaceState(application_manager::HmiInterfaces::HMI_INTERFACE_RC))
         .WillByDefault(Return(application_manager::HmiInterfaces::
                                   InterfaceState::STATE_AVAILABLE));
     ON_CALL(app_mngr_, application(kAppId)).WillByDefault(Return(mock_app_));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc]: @LitvinenkoIra second applicaiton ON_CALL available?

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc (hunk 2 above)
Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc]: @LitvinenkoIra may move it to SetuUp?

**Comment by LitvinenkoIra** [comment]
@LuxoftAKutsan fixed all comments 5c85449

**Pull request merged**